qutebrowser/misc/miscwidgets.py

```diff
@@ -93,9 +94,29 @@ def __on_cursor_position_changed(self, _old, new):
 
         We use __ here to avoid accidentally overriding it in subclasses.
         """
+        if self.hasSelectedText():
+            # Let __on_selection_changed handle it
+            return
+
         if new < self._promptlen:
             self.setCursorPosition(self._promptlen)
```

Comment:
Why not just replace this with `self.cursorForward(self.hasSelectedText(), self._promptlen - new)`?

Reply:
Completely remove it works fine too.